pyswmm/raingages.py

```diff
@@ -0,0 +1,241 @@
+    """
+
+    def __init__(self, model):
+        if not model._model.fileLoaded:
+            raise PYSWMMException("SWMM Model Not Open")
+        self._model = model._model
+        self._cuindex = 0
+        self._nRaingages = self._model.getProjectSize(ObjectType.GAGE.value)
+
+    def __len__(self):
+        """
+        Return number of raingages. Use the expression 'len(RainGages(sim))'.
+
+        :return: Number of Raingages
+        :rtype: int
+
+        """
+        return self._model.getProjectSize(ObjectType.GAGE.value)
+
+    def __contains__(self, raingageid):
+        """
+        Checks if Rain Gage ID exists.
+
+        :return: ID Exists
+        :rtype: bool
+        """
+        return self._model.ObjectIDexist(ObjectType.GAGE.value, raingageid)
+
+    def __getitem__(self, raingageid):
+        if self.__contains__(raingageid):
+            rg = RainGage(self._model, raingageid)
+            return rg
+        else:
+            raise PYSWMMException("Raingage ID: {} Does not Exist".format(raingageid))
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self._cuindex < self._nRaingages:
+            raingageobject = self.__getitem__(self._raingageid)
+            self._cuindex += 1  # Next Iteration
+            return raingageobject
+        else:
+            raise StopIteration()
+
+    next = __next__  # Python 2
+
+    @property
+    def _raingageid(self):
+        """Node ID."""
+        return self._model.getObjectId(ObjectType.GAGE.value, self._cuindex)
+
+
+class RainGage(object):
+    """
+    Raingage Methods.
+
+    :param object model: Open Model Instance
+    :param str raingageid: Raingage ID
+
+    Examples:
+
+    >>> from pyswmm import Simulation, Raingages
+    >>>
+    >>> with Simulation('tests/data/TestModel1_weirSetting.inp') as sim:
+    ...     rg1 = Raingages(sim)["Gage1"]
+    ...     print(rg1.raingageid)
+    ...     for step in simulation:
+    ...         print(rg1.total_precip)
+    ... Gage1
+    ... 0
+    ... 10
+    """
+
+    def __init__(self, model, raingageid):
+        if not model.fileLoaded:
+            raise PYSWMMException("SWMM Model Not Open")
+        if raingageid not in model.getObjectIDList(ObjectType.GAGE.value):
+            raise PYSWMMException("ID Not valid")
+        self._model = model
+        self._raingageid = raingageid
+
+    # --- Get Parameters
+    # -------------------------------------------------------------------------
+    @property
+    def raingageid(self):
+        """
+        Get Rain Gage ID.
+
+        :return: Parameter Value
+        :rtype: float
+
+        Examples:
+
+        >>> from pyswmm import Simulation, RainGages
+        >>>
+        >>> with Simulation('tests/data/TestModel1_weirSetting.inp') as sim:
+        ...     rg = RainGage(sim)["Gage1"]
+        ...     print(rg.raingageid)
+        >>> Gage1
+        """
+        return self._raingageid
+
+    @property
+    def total_precip(self):
```

Comment:
Does this change the precipitation rate of rain gage? total_precip sounds confusing to me for a rate (in/hr or mm/hr)

Reply:
@jennwuu, in my imagination we were going to hope people would read the documentation to know that this is the total precipitation rate (not cumulative).  Agree that the name is a little misleading, however total_precip = rainfall + snowfall rates

Reply:
would it make sense to put total_precip_rate or is that too long?

Reply:
@jennwuu, I think it's tooooo expressive to add that.